[case fastenum_forloop]

from fastenum import Enum

class Color(Enum):
    RED = 1
    GREEN = 2
    BLUE = 3


class Strings(Enum):
    A = 'a'
    B = 'b'
    C = 'c'


reveal_type(Color.__next__()) # E: Revealed type is 'main.Color'
reveal_type(Strings.__next__()) # E: Revealed type is 'main.Strings'

for color_v in Color:
	reveal_type(color_v) # E: Revealed type is 'main.Color'

for string_v in Strings:
	reveal_type(string_v) # E: Revealed type is 'main.Strings'

[/CASE]
